mpi: add new status_set/get functions (#13125)
open-mpi/ompi: docs/man-openmpi/man3/MPI_Status_get_error.3.rst

.. _mpi_status_get_error:

MPI_Status_get_error
========================

.. include_body

:ref:`MPI_Status_get_error` |mdash| Retrieves the MPI_ERROR field from *status*.

SYNTAX
------

C Syntax
^^^^^^^^

.. code-block:: c

   #include <mpi.h>

   int MPI_Status_get_error(MPI_Status *status, int *error)

Fortran Syntax
^^^^^^^^^^^^^^

.. code-block:: fortran

   USE MPI
   ! or the older form: INCLUDE 'mpif.h'
   MPI_STATUS_GET_ERROR(STATUS, ERROR, IERROR)
   	INTEGER	STATUS(MPI_STATUS_SIZE), IERROR
   	INTEGER ERROR

Fortran 2008 Syntax
^^^^^^^^^^^^^^^^^^^

.. code-block:: fortran

   USE mpi_f08
   MPI_Status_get_error(status, error, ierror)

>    	TYPE(MPI_Status), INTENT(INOUT) :: status

Reviewer: ```suggestion
   	TYPE(MPI_Status), INTENT(IN) :: status
```

Author: @bosilca thank you, yes, the INOUT is correct for the _set_error version of the functions, but not for the _get_ error. I've updated all _set versions to apply this change

   	INTEGER, INTENT(OUT) :: error
   	INTEGER, OPTIONAL, INTENT(OUT) :: ierror

> INPUT/OUTPUT PARAMETER

Reviewer: Why is the status inout ? It should be read only for get.

> INPUT/OUTPUT PARAMETER

Reviewer: ```suggestion
INPUTPARAMETER
```

Author: same comment as above (correct for the _set but not for the _get versions)

----------------------

> * ``status``: Status with which to retrieve error (status).

Reviewer: ```suggestion
* ``status``: Status from which to retrieve the error (status).
```

> INPUT PARAMETER

Reviewer: ```suggestion
OUTPUT PARAMETER
```

Reviewer 2: I don't think you just want to make this 1-line change -- I think that all 3 params should be merged into a single "OUTPUT" section.

Reviewer 2: (I deleted my previous comment)

Why do you suggest this?  Looking at MPI-4.1 p42:11, `status` is an IN parameter.

Reviewer: mistake I guess.

---------------
* ``error``: error to set in the MPI_ERROR field (integer).

OUTPUT PARAMETER
----------------
* ``ierror``: Fortran only: Error status (integer).

DESCRIPTION
-----------

> Returns in error the MPI_ERROR field in the status object.

Reviewer: ```suggestion
Returns in error the MPI_ERROR field from the status object.
```

Author: also same as above. "from" has to be used for the _get, functions, "in" for the _set functions.
